src/weapons/weapon.js

```diff
@@ -0,0 +1,26 @@
+import WeaponPlugin from 'phaser3-weapon-plugin';
+
+export class Weapon {
+
+  constructor(player) {
+    this.bullets = player.scene.weapons.add(30, 'bullet');
```

Comment:
I think this initially adds a weapon at x: 0, y: 0 and that this causes some big problems when we set a new bullets-worldLayer collider (there's constant collisions).

Reply:
Or maybe we use a different pattern.  WeaponPlugin provides lots of convenience methods for customising the weapon.  It's easy to change things like spread, fire rate, bullet icon, physics.  Maybe we just use some factories that produce weapons.